New easyconfig for DFT-D3

valtandor opened the pull request.

valtandor commented:
Yes.

boegel commented on easybuild/easyconfigs/d/DFT-D3/DFT-D3-3.1.1-intel-2015a.eb:
```diff
@@ -0,0 +1,21 @@
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+source_urls = ['http://www.thch.uni-bonn.de/tc/downloads/%(name)s/data']
+sources = ['dftd3.tgz']
```
for non-versioned tarballs, please include a checksum too

(or maybe even require to rename the tarball as well)

valtandor replied:
A checksum has now been added. I don't know how we could require that the tarball (especially if it is downloaded from a remote server) be renamed.

boegel replied:
We don't have support to automagically rename the tarball yet on download, but we should look into that.

For now, this will do.

boegel commented:
Jenkins: ok to test

boegel commented on easybuild/easyconfigs/d/DFT-D3/DFT-D3-3.1.1-intel-2015a.eb:
```diff
@@ -0,0 +1,22 @@
+
+source_urls = ['http://www.thch.uni-bonn.de/tc/downloads/%(name)s/data']
+sources = ['dftd3.tgz']
+checksums = [('md5', '88470bd398b5cc90ad680ed1df4239f7')]
```
I'm now getting a different checksum, so was a new version released?

I'd suggest to use `dftd3-%(version)s.tgz` as source file name, implying that it should be downloaded manually and renamed (until we have support for automagically renaming on download in the framework), since this is not going to work out well otherwise, clearly...

valtandor commented:
@boegel: I've added the version decoration in the name, added a comment and taken out checksum verification (the authors of the software unfortunately aren't above stealth upgrades). Let me know if you would like further changes.

boegel commented:
Jenkins: test this please

boegel commented on easybuild/easyconfigs/d/DFT-D3/DFT-D3-3.1.1-intel-2015a.eb:
```diff
@@ -0,0 +1,27 @@
+# reason, it is suggested to manually download and rename the tarball, and the
+# checksum input has been disabled.
+sources = ['dftd3-%(version)s.tgz']
+#checksums = [('md5', '88470bd398b5cc90ad680ed1df4239f7')]
```
the comment above is actually a reason to keep the checksum in place...

how else can EB be sure that the source tarball you're using matches the version specified?

boegel replied:
w.r.t. the stealth upgrades: if it happens, we want to know about it, and the checksum should be updated then

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/18db6959cdad36a46fd8 for a full test report.

boegel commented on easybuild/easyconfigs/d/DFT-D3/DFT-D3-3.1.1-intel-2015a.eb:
```diff
@@ -0,0 +1,27 @@
+# reason, it is suggested to manually download and rename the tarball. The
+# checksum may also need updating from time to time.
+sources = ['dftd3-%(version)s.tgz']
+checksums = [('md5', '88470bd398b5cc90ad680ed1df4239f7')]
```
hmm, I'm getting `5d1cbaa851cd88ae515dea3cc9a7590d`

and it still seems to be v3.1.1?

valtandor replied:
@boegel: Yes, and it looks as though it was last updated on 11 January. Probably another stealth upgrade...sigh.

boegel replied:
Time to reach out to them?

boegel commented:
@valtandor: this can't be merged until the checksum is updated?

valtandor commented:
@boegel I've just updated the checksum, but I don't know that it won't change again (though my OS reports that the tarball was last changed on 11 January, so hopefully it's sort of stable).

boegel commented:
lgtm

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6e84b88994db485fb5f4c81238043fbd for a full test report.

boegel commented:
Going in, thanks @valtandor!
